assignments/haskell/matrix/matrix_test.hs

```diff
@@ -0,0 +1,78 @@
+import Test.HUnit (Assertion, (@=?), runTestTT, Test(..))
+import Control.Monad (void)
+import Matrix ( Matrix, row, column, rows, cols, shape, transpose
+              , reshape, flatten, fromString, fromList)
+import Control.Arrow ((&&&))
+import qualified Data.Vector as V
+
+-- Implementation of a row-major matrix for any type, using Data.Vector.
+
+-- fromString should work for any type that implements the Read typeclass.
+-- You can assume that "\n" is used to delimit rows without context
+-- sensitivity, but treatment of whitespace in a row should be context
+-- sensitive!
+
+-- No validation of input is required. Let it fail if the matrix is not
+-- rectangular, invalid chars are encountered, etc.
+
+-- shape is (rows, cols)
+
+testCase :: String -> Assertion -> Test
+testCase label assertion = TestLabel label (TestCase assertion)
+
+main :: IO ()
+main = void $ runTestTT $ TestList
+       [ TestList matrixTests ]
+
+v :: [a] -> V.Vector a
+v = V.fromList
+
+msInt :: String -> Matrix Int
+msInt = fromString
+
+msInteger :: String -> Matrix Integer
+msInteger = fromString
+
+msChar :: String -> Matrix Char
+msChar = fromString
+
+msString :: String -> Matrix String
+msString = fromString
+
+matrixTests :: [Test]
+matrixTests =
+  [ testCase "extract first row" $ do
+    v [1, 2] @=? row 0 (msInt "1 2\n10 20")
+    v [9, 7] @=? row 0 (msInteger "9 7\n8 6")
+  , testCase "extract second row" $ do
+    v [19, 18, 17] @=? row 1 (msInt "9 8 7\n19 18 17")
+    v [16, 25, 36] @=? row 1 (msInteger "1 4 9\n16 25 36")
+  , testCase "extract first column" $ do
+    v [1, 4, 7, 8] @=? column 0 (msInt "1 2 3\n4 5 6\n7 8 9\n 8 7 6")
```

Comment:
oh noes, an extra space. let us remove it in https://github.com/exercism/haskell/pull/651